### PR TITLE
Xonline tweaks

### PR DESCRIPTION
**Pull request by LukeUsher**
Included in this pull-request:
Partial emulation of XOnline libraries
Implementation of more X86 opcodes
Kernel bug fixes

The end result: Titles that link with XOnline and related libraries are less likely to crash.

**Newly Working Titles (Promoted to Menus status)**

Xbox Live Dashboard:
image
image

Conflict Desert Storm III (Pal):
image

### CODE REVIEW

File: src/CxbxKrnl/EmuNV2A.cpp [hunk 1]
```diff
@@ -1235,28 +1236,68 @@ const NV2ABlockInfo* EmuNV2A_Block(xbaddr addr)
...
+		uint32_t aligned_value = 0;
+		uint32_t mask = 0;
+		switch (size) {
+			case 8:
+				shift = (addr & 3) * 8;
+				aligned_addr = addr & ~3;
+				aligned_value = block->read(aligned_addr - block->offset);
+				mask = 0xFF << shift;
+
+				// TODO : Must the second byte be written to the next DWORD?
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuNV2A.cpp]: in this case there is no second byte

**Comment by ObiKKa** [comment]
Actually your title above the third screenshot in the OP has a little typo: The game is "Conflict Desert Storm II" (2nd instalment), not "Conflict Desert Storm III (Pal)"

**Comment by PatrickvL** [comment]
Great work, thanks!

File: src/CxbxKrnl/EmuNV2A.cpp (hunk 1 above)
Comment (LukeUsher) [inline src/CxbxKrnl/EmuNV2A.cpp]: Thanks, that's what I get for copying and pasting a little too hastily

**Pull request merged**